docs/DeveloperGuide.md

```diff
@@ -1,9 +1,263 @@
+If the project name is not given, such as in the following command, all projects will be searched to find resources matching the keyword. 
+> find k/website
+
+The following sequence diagram shows how the find operation works:
+![Exit_puml](puml_img/Find.png)
+
+**Design Consideration**
+
+Aspect: How find executes
+* Alternative 1 (current choice):
+  * Allow the user to input find k/KEYWORD p/PROJECTNAME or find k/KEYWORD 
+    based on whether he wants to find relevant resources in a single project or all projects.
+  * **Pros**: Fewer commands for the user to choose between. Simple to implement.
+* Alternative 2:
+  * Allow the user to input 2 separate commands find p/PROJECTNAME k/KEYWORD and find-all k/KEYWORD 
+    based on whether he wants to find relevant resources in a single project or all projects.
+  * **Pros**: Simple to implement.
+  * **Cons**: More functions and code required.
+
+### Exiting TraceYourProj
+**Proposed Implementation**
+
+Given below is an example usage scenario and how the exit mechanism behaves at each step.
+
+**Step 1**:
+The user initialises the application and adds the resources, Project and Resource objects are created accordingly.
+
+**Step 2**:
+The user has finished using the application and wishes to exit the application. He executes the following command to leave the application. 
+> exit
+
+The following sequence diagram shows how the exit operation works:
+![Exit_puml](puml_img/Exit.png)
+
+**Design Consideration**
+
+Aspect: How exit executes
+* Alternative 1 (current choice):
+  * Break out of do-while loop in the main function in Duke when boolean condition isLoop 
+    is no longer true.
+  * **Pros**: Easy to implement.
+  * **Cons**: Possibility of entering an infinite loop if not properly coded.
+* Alternative 2:
+  * Execute System.Exit(0) when exit is executed.
+  * **Pros**: Easy to implement.
+  * **Cons**: Reduces testability since more code is required to throw an exception when System.exit is 
+    called and catch that exception, so the JUnit test does not fail.
+
+## Saving and loading data
+
+**Proposed Implementation**
+
+Given below is an example usage of how the saving and loading functionality works.
+
+**Step 1:**
+The user initialises the application and adds the resources, Project and Resource objects are created accordingly.
+
+**Step 2:**
+The user issues the save command by typing “save”
+
+**Step 3:**
+The Project and Resource objects contents are saved to a text file upon exiting the program
+
+**Step 4:**
+The application state can be restored by the Storage class which can read the text file and creates the objects 
+to create an application state when the program starts by issuing the load command by typing “load”
+
+The following sequence diagram shows how the **save** operation works:
+![Save_puml](puml_img/Save.png)
+
+The following sequence diagram shows how the **load** operation works:
+![Load_puml](puml_img/Load.png)
```

Comment:
It would be good if you could explain what happens if there is no text file to read from.